dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/program/notification/ProgramNotificationInstanceServiceTest.java

```diff
@@ -107,9 +107,24 @@ void shouldReturnProgramNotificationInstancesForGivenDate() {
     assertEquals(
         expectedNotifications,
         instances.size(),
-        "Expected " + expectedNotifications + " notification instances, but got " + instances);
+        () ->
+            String.format(
+                "Expected %d notification instances, but got %d.\nNo instances were found for scheduledAt: %s."
```

Comment:
`No instances were found for scheduledAt: %s`

Is that true if you expect 20 and get 19 or so?